web/thesauruses/javascript/ECMAScript 2023/strings.json

```diff
@@ -0,0 +1,352 @@
+            "str.search(substring)"
+        ]
+      },
+      "count_occurrences_of_substring": {
+        "name": "Find number of occurences of a substring",
+        "code": [
+            "str.split(substring).length - 1",
+            "str.match(/substring/g).length"
+        ]
+      },
+      "get_leftmost_characters": {
+        "name": "Get a specified number of characters from the left",
+        "code": [
+            "str.slice(start, end)",
+            "str.substring(start, end)",
+            "str.substr(start, length)"
+        ]
+      },
+      "get_rightmost_characters": {
+        "name": "Get a specified number of characters from the right",
+        "code": [
+            "str.slice(-end, -start)",
+            "str.substring(str.length - end, str.length - start)",
+            "str.substr(str.length - length, length)"
+        ]
+      },
+      "get_substring_from_start_and_end_index": {
+        "name": "Return a substring from a string based on starting and ending indices",
+        "code": [
+            "str.slice(start, end)",
+            "str.substring(start, end)",
+            "str.substr(start, length)"
+        ]
+      },
+      "get_substring_from_start_index_and_length": {
+        "name": "Return a substring from a string based on starting index and size of substring",
+        "code": [
+            "str.slice(start, start + length)",
+            "str.substring(start, start + length)",
+            "str.substr(start, length)"
+        ]
+      },
+      "convert_to_uppercase": {
+        "name": "Convert string to all uppercase",
+        "code": "str.toUpperCase()"
+      },
+      "convert_to_lowercase": {
+        "name": "Convert string to all lowercase",
+        "code": "str.toLowerCase()"
+      },
+      "convert_to_title_case": {
+        "name": "Convert string to title case",
+        "code": "str.replace(/\\w+/g, (w) => w[0].toUpperCase() + w.slice(1).toLowerCase())"
+      },
+      "capitalize_string": {
+        "name": "Capitalize first letter of a string",
+        "code": [
+            "str[0].toUpperCase() + str.slice(1)",
+            "str.replace(/^\\w/, (w) => w.toUpperCase())"
+        ]
+      },
+      "remove_whitespace": {
+        "name": "Remove all whitespaces from string",
+        "code": "str.replace(/\\s/g, \"\")"
+      },
+      "replace_substring": {
+        "name": "Replace a substring with another string",
+        "code": [
+            "str.replace(substring, another)",
+            "str.split(substring).join(another)"
+        ]
+      },
+      "replace_all_substring": {
+        "name": "Replace all substring matches with another string",
+        "code": [
+            "str.replace(/substring/g, another)",
+            "str.split(substring).join(another)"
+        ]
+      },
+      "split_at_index": {
+        "name": "Split string into a list of strings at a given index",
+        "code": "str.split(separator, limit)",
+        "comment": ""
```

Comment:
Did you mean to leave an empty comment?

Reply:
I'll remove that (if I have no comment to write there)